### PR TITLE
Fix binary stats for arrays containing null bytes and match stats behaviour between varbin and primitive arrays

### PR DESCRIPTION
**Pull request by robert3005**
Made the bevaiour match the primitive stats as well where min/max is null if whole array is null

### CODE REVIEW

File: vortex-array/src/array/varbin/stats.rs [hunk 1]
```diff
@@ -44,15 +75,15 @@ pub struct VarBinAccumulator<'a> {
 }
 
 impl<'a> VarBinAccumulator<'a> {
-    pub fn new() -> Self {
+    pub fn new(value: Cow<'a, [u8]>) -> Self {
         Self {
-            min: Cow::from(&[0xFF]),
-            max: Cow::from(&[0x00]),
```
Comment (robert3005) [inline vortex-array/src/array/varbin/stats.rs:50]: while these values worked for utf8 they didn't work for binary

File: vortex-array/src/array/primitive/stats.rs [hunk 2]
```diff
@@ -38,15 +38,15 @@ impl<T: NativePType> StatsCompute for &[T] {
...
 
-fn all_null_stats<T: NativePType>() -> VortexResult<StatsSet> {
+fn all_null_stats<T: NativePType>(len: usize) -> VortexResult<StatsSet> {
     Ok(StatsSet::from(HashMap::from([
         (Stat::Min, Option::<T>::None.into()),
         (Stat::Max, Option::<T>::None.into()),
         (Stat::IsConstant, true.into()),
         (Stat::IsSorted, true.into()),
-        (Stat::IsStrictSorted, true.into()),
+        (Stat::IsStrictSorted, (len == 1).into()),
```
Comment (lwwmanning) [inline vortex-array/src/array/primitive/stats.rs]: len < 2

?

File: vortex-array/src/array/primitive/stats.rs [hunk 3]
```diff
@@ -38,15 +38,15 @@ impl<T: NativePType> StatsCompute for &[T] {
...
         (Stat::RunCount, 1.into()),
```
Comment (lwwmanning) [inline vortex-array/src/array/primitive/stats.rs:48]: len > 0

?

File: vortex-array/src/array/varbin/stats.rs [hunk 4]
```diff
@@ -15,22 +16,52 @@ impl StatsCompute for VarBinArray {
...
+    }
+}
+
+fn all_null_stats(len: usize, dtype: &DType) -> StatsSet {
+    StatsSet::from(HashMap::from([
+        (Stat::Min, Scalar::null(dtype)),
+        (Stat::Max, Scalar::null(dtype)),
+        (Stat::IsConstant, true.into()),
+        (Stat::IsSorted, true.into()),
+        (Stat::IsStrictSorted, (len == 1).into()),
```
Comment (lwwmanning) [inline vortex-array/src/array/varbin/stats.rs]: Same as above

File: vortex-array/src/array/primitive/stats.rs (hunk 3 above)
Comment (robert3005) [reply vortex-array/src/array/primitive/stats.rs:48]: Len == 0 is handled separately

**Review by lwwmanning — APPROVED** [review]

**Pull request merged**